torchaudio/transforms.py

```diff
@@ -1730,242 +1703,100 @@ def __init__(
-            reference_vector (torch.Tensor): one-hot reference channel matrix
-
-        Returns:
-            torch.Tensor: the enhanced STFT
-                Tensor of dimension `(..., freq, channel, 1)`
-        """
-        phi = torch.linalg.solve(psd_n, psd_s)  # psd_n.inv() @ psd_s
-        stv = torch.einsum("...fec,...c->...fe", [phi, reference_vector])
-        stv = stv.unsqueeze(-1)
-        stv = torch.matmul(phi, stv)
-        stv = torch.matmul(psd_s, stv)
-        return stv
-
-    def _apply_beamforming_vector(self, specgram: torch.Tensor, beamform_vector: torch.Tensor) -> torch.Tensor:
-        r"""Apply the beamforming weight to the noisy STFT
-        Args:
-            specgram (torch.tensor): multi-channel noisy STFT
-                Tensor of dimension `(..., channel, freq, time)`
-            beamform_vector (torch.Tensor): beamforming weight matrix
-                Tensor of dimension `(..., freq, channel)`
-
-        Returns:
-            torch.Tensor: the enhanced STFT
-                Tensor of dimension `(..., freq, time)`
-        """
-        # (..., channel) x (..., channel, freq, time) -> (..., freq, time)
-        specgram_enhanced = torch.einsum("...fc,...cft->...ft", [beamform_vector.conj(), specgram])
-        return specgram_enhanced
-
-    def _tik_reg(self, mat: torch.Tensor, reg: float = 1e-7, eps: float = 1e-8) -> torch.Tensor:
-        """Perform Tikhonov regularization (only modifying real part).
-        Args:
-            mat (torch.Tensor): input matrix (..., channel, channel)
-            reg (float, optional): regularization factor (Default: 1e-8)
-            eps (float, optional): a value to avoid the correlation matrix is all-zero (Default: 1e-8)
-
-        Returns:
-            torch.Tensor: regularized matrix (..., channel, channel)
-        """
-        # Add eps
-        C = mat.size(-1)
-        eye = torch.eye(C, dtype=mat.dtype, device=mat.device)
-        with torch.no_grad():
-            epsilon = _get_mat_trace(mat).real[..., None, None] * reg
-            # in case that correlation_matrix is all-zero
-            epsilon = epsilon + eps
-        mat = mat + epsilon * eye[..., :, :]
-        return mat
-
-    def forward(
-        self, specgram: torch.Tensor, mask_s: torch.Tensor, mask_n: Optional[torch.Tensor] = None
-    ) -> torch.Tensor:
+    def forward(self, specgram: Tensor, mask_s: Tensor, mask_n: Optional[Tensor] = None) -> Tensor:
         """Perform MVDR beamforming.
 
         Args:
-            specgram (torch.Tensor): the multi-channel STF of the noisy speech.
+            specgram (Tensor): the multi-channel STF of the noisy speech.
                 Tensor of dimension `(..., channel, freq, time)`
-            mask_s (torch.Tensor): Time-Frequency mask of target speech.
+            mask_s (Tensor): Time-Frequency mask of target speech.
                 Tensor of dimension `(..., freq, time)` if multi_mask is ``False``
```

Comment:
I think perhaps you should comment on the expected type and/or range of mask_s and mask_n.  E.g. are they expected to be boolean or reals in [0..1]; and are they expected to sum to one?  Also perhaps comment on how
they would typically be obtained.  This may help those (such as myself!) who are not super familiar with 
beamforming issues.
You might also want to specify that mask_s and mask_n cannot be zero everywhere (if indeed that is the case).

Reply:
Sometimes the sum of `mask_s` and `mask_n` will not be one, for example, the [phase-sensitive mask](https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.727.7832&rep=rep1&type=pdf) or the masks directly estimated by the neural networks. But you are right, in this implementation they must be real instead of boolean.

Sure, I will update the docstring to reflect those, thanks!

Reply:
> You might also want to specify that mask_s and mask_n cannot be zero everywhere (if indeed that is the case).

This is partially correct. Normally, the necessary property is, that the mask cannot be zero everywhere for any frequency. So on each frequency you need one non-zero value.
The implementation on the other side, uses some regulations, that the code will not fail.

In my experience, I ensure that the mask has this property and drop the regulations.

Reply:
In neural-network based training, the mask values can be very small for some frequency bin, considering that I will keep the regulation term and relax the non-zero check.

Reply:
I know. I observed that `mask + 0.01` is a good alternative regulation and has the advantage to be scale independent.
Not sure, but I compared it once and saw a tiny gain for this, so I kept this in my experiments.